agent/ui_endpoint.go

```diff
@@ -17,26 +17,27 @@ import (
 const metaExternalSource = "external-source"
 
 type GatewayConfig struct {
-	Addresses []string `json:",omitempty"`
+	AssociatedServiceCount int      `json:",omitempty"`
+	Addresses              []string `json:",omitempty"`
 	// internal to track uniqueness
 	addressesSet map[string]struct{}
 }
 
 // ServiceSummary is used to summarize a service
 type ServiceSummary struct {
-	Kind              structs.ServiceKind `json:",omitempty"`
-	Name              string
-	Tags              []string
-	Nodes             []string
-	InstanceCount     int
-	ProxyFor          []string            `json:",omitempty"`
```

Comment:
Does the UI use this information for anything? Is it ok to get rid of this and instead only return a boolean?

Reply:
It's only used for the `connected with proxy` label